### PR TITLE
Fix admin images with Django 1.9 and multi usage of file picker widget

### PR DESCRIPTION
**Pull request by acatton**
This pull request is kind of a melting pot. I just wanted to fix the static files in the admin center for Django 1.9. But I ended up cleaning up some code as well, and fixing some other things.

 - [x] Fix the images for the FileField() in the admin center
 - [x] Fix the fragility of the javascript that handles static files
 - [x] Fix the confusing names `DJANGO_X_X` in `utils/compatibility.py`.
 - [x] Fix the last bits of code that weren't using `LTE_DJANGO_X_X`
 - [x] Fix file picket widget failing in inlines (#683 / #743)
 - [x] Fix file picket widget failing when used multiple times on the same form

**Edit**: I would strongly advised to review each commit one by one instead of reviewing the whole thing.

### CODE REVIEW

**Comment by joshuajonah** [comment]
+1

**Comment by mkoistinen** [comment]
@acatton This looks like great stuff, thanks. What is the current status of this PR?

**Comment by Flight** [comment]
FE LGTM

**Comment by mkoistinen** [comment]
@acatton Also, can you rebase this against latest develop please?

**Comment by acatton** [comment]
@mkoistinen, @julianandrews or @gavinwahl is supposed to take over this pull request.

**Comment by julianandrews** [comment]
@mkoistinen, @Flight  I've cherry picked the relevant commits from #683 and rebased this PR on top of develop. I decided to make some minor changes to the logic from 3f7e6101 to mesh better with the `createDropzone` function @acatton made, but other than that, to the best of my ability this is just a straightforward rebase of the changes in this PR and #683.

**Comment by gavinwahl** [comment]
Hey @mkoistinen -- this was rebased a couple weeks ago, but now changes to develop have broken again. This is the 3rd or 4th time we've had to do this. Can we get some kind of confirmation that this will be merged immediately if we do it again?

Thanks

File: filer/static/filer/js/addons/popup_handling.js [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
-        var addFolderButton = $('#' + id);
-
-        input.val(chosenId);
-        folderName.text(chosenName);
-        clearButton.removeClass('hidden');
-        addFolderButton.addClass('hidden');
-        win.close();
-    };
-})(django.jQuery);
+<<<<<<< HEAD
```
Comment (vstoykov) [inline filer/static/filer/js/addons/popup_handling.js]: Here is left a line from merging.

**Comment by stefanfoulis** [comment]
@acatton @julianandrews this PR is awesome and fixes much more than the title indicates. I took the liberty to change the title and updated the description to reflect everything it fixes. I'd like to merge it asap. I you rebase it again, I'll merge it right away. If you don't have time I'll try to tackle the rebase tomorrow morning (CET).
Thanks for the awesome work @acatton and sorry it took us so long to get it merged.

sidenote: does the js refactor happen to also address #736 (clash is other widgets on the same page use jQuery)?

**Comment by acatton** [comment]
@stefanfoulis full disclosure, I left the company and the project that was extensively using this. I was hoping to leave with this pull request being merged, but it didn't work out :wink:  . This is why I might have been a _little_ passive aggressive in my comments towards the end. And I would like to apologize for this :sweat_smile:  . No worries, @julianandrews took over the project in question and this pull request as well.

I see that julian merged the two pull requests. I didn't do that because I felt that they were unrelated work, even though #683 was based on this. But julian's the boss now.

What this does is making the javascript a lot more robust. It's difficult to explain, but it is relying on the structure on the page more than on some internal state held in a javascript variable. To answer your question, @julianandrews would have to test it (because I don't have the setup to test it anymore) but it could very likely also fix #736. But i wouldn't like to set these expectations too high.

**Comment by stefanfoulis** [comment]
@acatton thanks for the quick response. Hopefully @julianandrews can do the rebase. If not, that fine as well and I'll give it a shot. Anyhow, the help of everyone involved is much appreciated :+1: :trophy:

**Comment by julianandrews** [comment]
I think this should do it. This at least seems to fix #683. I'm not sure if it addresses #736.

#743 seems to be unfixed; I'm not sure if that's somehow the result of the rebase. I have a good idea at this point how to go about fixing #743, so I'll take a look, but if possible I'd rather get this PR merged in, and I'll submit a much smaller PR if I come up with a good fix.

**Comment by julianandrews** [comment]
@stefanfoulis 

Ignore my last comment about #743. This *does* fix #743. Somehow I had a cached copy of `admin/js/inlines.js` in my browser that wasn't creating the appropriate event. This should just work!

**Comment by stefanfoulis** [comment]
awesome @julianandrews . Thank you very much for the quick response and rebase. It's late at night here, so I'll postpone the review and merge to tomorrow morning.

**Pull request merged**